docs/01-readme.md

```diff
@@ -52,7 +52,26 @@ make package
 sudo apt install *.deb
 ```
 
-
+#### NixOS (overlay in configuration.nix)
+```nix
+# Overlay bellow will build AwesomeWM from selected commit
+# NOTE: Manually change revision (commit numbers) on every update.
+
+nixpkgs.overlays = [ 
+(self: super: {
+  awesome = super.awesome.overrideAttrs (old: {
+    src = super.fetchFromGitHub {
+      owner = "awesomewm";
+      repo = "awesome";
+      rev = "4319b161103a81403ba3516924e86e13ff33c036";
```

Comment:
This is interesting (quite different from what ever saw). Cannot the 'rev' attribute be something like tag name or branch name ? This seems like you're pointing to current 'master'

I'm asking because I'm guessing Nix uses `git` to download the repo. `git` does support fetching a branch

Reply:
I tried without "revision" but it gives mi error about missing "rev" function and cannot build it. ``error: anonymous function at /nix/store/7579j58gq0v5jwm3d25b9wfz2nwbc3a6-nixos-21.05pre271938.c7d0dbe094c/nixos/pkgs/build-support/fetchgithub/default.nix:3:1 called without required argument 'rev', at /etc/nixos/modules/display/X11.nix:10:11``

Reply:
ok, seems like you can pass it a tag name according to this [site](https://rawkode.com/articles/contributing-nixpkgs/), not really bothers me